libs/vm-modules/src/math/bignumber.cpp

```diff
@@ -250,6 +258,116 @@ bool UInt256Wrapper::FromJSON(JSONVariant const &variant)
   return true;
 }
 
+void UInt256Wrapper::Add(fetch::vm::Ptr<Object> &lhso, fetch::vm::Ptr<Object> &rhso)
+{
+  Ptr<UInt256Wrapper> lhs = lhso;
+  Ptr<UInt256Wrapper> rhs = rhso;
+  if (lhs->IsTemporary())
+  {
+    lhs->number_ += rhs->number_;
+    return;
+  }
+  if (rhs->IsTemporary())
+  {
+    rhs->number_ += lhs->number_;
+    lhso = std::move(rhs);
+    return;
+  }
+  Ptr<UInt256Wrapper> n(
+      new UInt256Wrapper(vm_, fetch::vm::TypeIds::UInt256, lhs->number_ + rhs->number_));
```

Comment:
Constructor doesn't need to be passed fetch::vm::TypeIds::UInt256 here. The constructor should pass TypeIds::UInt256 to the Object constructor itself.

Reply:
There is no constructor with the proper typeId in UInt256Wrapper, all constructors take it as a parameter. I guess it is better to pass here a type id instead of creation of a specialized constructor only for using it in 4 math operations inside this file.

Reply:
I found that a constructor that takes Id is needed in order to work with the recent changes in StructuredData, that way GetObject/SetObject can call a generic constructor. I had to do that for Fixed128 also, so it's ok.

Reply:
Having said that, for within the class itself, there is no point to carry TypeId everywhere, the Constructor with TypeId as argument is only useful within StructureData serialization/deserialization. Within the class itself, you could just call a plain constructor.

Reply:
VM uses a constructor with type_id as an argument
libs/vm/include/vm/vm.hpp:353:23: 
    return Ptr<T>{new T(this, GetTypeId<T>(), std::forward<Ts>(args)...)};
so I have just added another utility constructor just for such use-cases.